### ISSUE TITLE
Control Flow.

### ISSUE THREAD

**Issue by mathisonian** [idyll-lang/idyll#275]
For supporting data-driven reports better we probably need to support mapping over data to output idyll components.

I'd suggest something like this (given input `data = [1, 2, 3 ]`

```

normal text

[each data:d]
   [display var:d /]
[/each]

more normal text

```

should output

```
normal text

1
2
3 

more normal text
```

- [x] - Conditional
- [x] - Switch
- [ ] - Map

**Comment by mathisonian** [comment]
I initially resisted the urge to put control flow types of things into the language definition, and indeed I think the component API is rich enough now that we can generate performant responsive control flow without introducing additional syntax. 

I've used version of these components throughout various projects, and while they are relatively small and simple to implement, the task of doing it yourself each time feels like an unnecessary chore. I'd propose adding the following control flow components:

_conditional_
```
[Conditional if:myVar]
[/Conditional]
```

_switch_
```
[Switch value:myVar ]
   [Case test:"string1"] Content 1 [/Case]
   [Case test:"string2"] Content 2 [/Case]
[/Switch]
```

_map_

```
[Map value:myArrayVar currentValue:iterator ]
  [Item value:iterator /]
[/Map]
```

**Comment by mathisonian** [comment]
`Conditional` was added in https://github.com/idyll-lang/idyll/pull/394

**Comment by codemayank** [comment]
@mathisonian I can take this one and add Switch and Map Components.

**Comment by mathisonian** [comment]
This has been addressed by #524 and #528
